lib/Controller/SAMLController.php

```diff
@@ -85,13 +85,34 @@ public function __construct($appName,
 	 */
 	private function autoprovisionIfPossible(array $auth) {
 		$uidMapping = $this->config->getAppValue('user_saml', 'general-uid_mapping');
+		$displaynamefirstpartMapping = $this->config->getAppValue('user_saml','general-displaynamefirstpart_mapping');
+		$displaynamesecondpartMapping = $this->config->getAppValue('user_saml','general-displaynamesecondpart_mapping');
```

Comment:
I personally would prefer to not separate between first name and second name here. Rather just a single input field that can take multiple fields. (such as `{{fieldBar}} {{fieldFoo}}`) 

If you can give me commit access to your fork I can try to come up with something like that later.